wp-admin/js/customize-controls-site-icon-maskable.js

```diff
@@ -28,8 +29,59 @@ wp.customize(
 				// Set initial active state.
 				updateActive();
 
+				/**
+				 * Validate site icons for its presence and size.
+				 */
+				const iconValidation = () => {
+					const iconData = wp.customize
+						.control('site_icon')
+						.container.find('img.app-icon-preview');
+					const NotificationData = {
+						dismissible: true,
+						message: '',
+						type: 'error',
+						code: null,
+					};
+					const notifications = [];
+
+					if (!iconData.length) {
+						notifications.push(
+							new wp.customize.Notification('pwa_icon_not_set', {
+								...NotificationData,
+								message: PWA_IconMessages.pwa_icon_not_set,
+							})
+						);
+					}
+
+					if (
+						iconData.length &&
+						(iconData[0].naturalHeight < 512 ||
+							iconData[0].naturalHeight < 512)
+					) {
+						notifications.push(
+							new wp.customize.Notification(
+								'pwa_icon_too_small',
+								{
+									...NotificationData,
+									message:
+										PWA_IconMessages.pwa_icon_too_small,
+								}
+							)
+						);
+					}
+
+					notifications.map((notification) => {
+						wp.customize
+							.section('title_tagline')
```

Comment:
Since the control can be moved to another section, it would be best to get the section via `siteIconControl.section()`:

```suggestion
							.section(siteIconControl.section())
```

Reply:
Nevertheless, the notification should rather be added to the _control_ instead of being added to the _section_.

Reply:
A key reason is that the top of the section may be scrolled out of view, so the user may not see it.